### PR TITLE
Improvements in Inference manager, problem type and VGG bug fix.

### LINKED ISSUE #167: Improvement in Inference Manager for Classification Tasks
**Is your feature request related to a problem? Please describe.**
It is related to the tasks including K-fold cross validation.

**Describe the solution you'd like**
Inference manager should be improved by looping over the number of folds, and in case it is a classification problem, the results should be averaged over the folds and saved as probabilities.csv file.

**Additional context**

### PR DESCRIPTION
**Pull request by orhunguley**
Fixes #167, #168 and #169
## Proposed Changes

Inference manager is improved by looping over the number of folds, and in case it is a classification problem, the results will averaged over the folds and saved as probabilities.csv file.
- The final layer specified in the config file will be used in forward pass of VGG.
- find_problem_type() function return classification if final_layer in ["classification_but_not_softmax", "logits",  "classification_without_softmax"].

## Checklist

- [x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [x] Non-breaking change (would not break existing functionality): if changes breaks current code, please provide as many details as possible
- [x] Function/class source code documentation added/updated
- [x] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [ ] If adding a submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [ ] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [ ] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [ ] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation

### CODE REVIEW

**Comment by sarthakpati** [comment]
This all looks good, but the fact that test coverage is going down by **this** much is not acceptable, and thus I will need you to add unit tests to cover these changes.

**Review by sarthakpati — CHANGES_REQUESTED** [review]
This all looks good, but the fact that test coverage is going down by **this** much is not acceptable, and thus I will need you to add unit tests to cover these changes.

File: testing/config_classification.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ model:
   final_layer: softmax
 nested_training:
   testing: -2
-  validation: -2
+  validation: 3
```
Comment (sarthakpati) [inline testing/config_classification.yaml:25]:
```suggestion
  validation: -3
```

Single round of validation, otherwise the tests take too long.

Comment (orhunguley) [reply testing/config_classification.yaml:25]: I see, but  I think the coverage is decreased to lack of n-folds. Would keeping it as 2 be okay?

File: testing/test_full.py [hunk 2]
```diff
@@ -425,6 +425,45 @@ def test_inference_classification_rad_3d(device):
...
+    # read and initialize parameters for specific data dimension
+    parameters = parseConfig(
+        testingDir + "/config_classification.yaml", version_check=False
+    )
+    parameters["modality"] = "rad"
+    parameters["patch_size"] = patch_size["3D"]
+    parameters["model"]["dimension"] = 3
+    parameters["model"]["amp"] = True
+    parameters["model"]["final_layer"] = "logits"
+    parameters["nested_training"]["validation"] = 2
```
Comment (sarthakpati) [inline testing/test_full.py:438]: Need to ensure that the inference works when even a single validation fold is presented.

File: testing/config_classification.yaml (hunk 1 above)
Comment (sarthakpati) [reply testing/config_classification.yaml]: Add both cases in the test.

File: testing/test_full.py (hunk 2 above)
Comment (orhunguley) [reply testing/test_full.py]: both single and multi fold cases should be covered with the latest commit

**Review by sarthakpati — APPROVED** [review]
LGTM!
